packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,514 @@
+  private readonly requiredSigners: number;
+  private readonly totalSigners: number;
+  private readonly changeOutputIndices: number[];
+
+  /**
+   * Constructor for RbfTransaction
+   *
+   * @param {RbfTransactionOptions} options - Configuration options for the RBF transaction
+   * @throws {Error} If the PSBT is not in a valid format or not signaling RBF
+   */
+  constructor(options: RbfTransactionOptions) {
+    this.originalPsbt = initializePsbt(options.psbt);
+    this.newPsbt = new PsbtV2();
+    this.originalPsbt.copy(this.newPsbt);
+    this.network = options.network;
+    this._targetFeeRate = options.targetFeeRate;
+    this.feeOutputIndex = options.feeOutputIndex;
+    this.dustThreshold = new BigNumber(
+      options.dustThreshold || DEFAULT_DUST_THRESHOLD,
+    );
+    this.additionalUtxos = options.additionalUtxos || [];
+    this.requiredSigners = options.requiredSigners;
+    this.totalSigners = options.totalSigners;
+    this.changeOutputIndices = options.changeOutputIndices;
+    this.validatePsbt();
+  }
+
+  private validatePsbt(): void {
+    if (!this.newPsbt.isRBFSignaled) {
+      throw new Error("This transaction is not signaling RBF.");
+    }
+  }
+
+  // getters
+
+  get psbt(): PsbtV2 {
+    return this.newPsbt;
+  }
+
+  get targetFeeRate(): FeeRateSatsPerVByte {
+    return this._targetFeeRate;
+  }
+
+  get totalInputValue(): string {
+    return calculateTotalInputValue(this.newPsbt).toString();
+  }
+
+  get totalOutputValue(): string {
+    return calculateTotalOutputValue(this.newPsbt).toString();
+  }
+
+  get currentFee(): string {
+    return new BigNumber(this.totalInputValue)
+      .minus(this.totalOutputValue)
+      .toString();
+  }
+
+  get requiredFee(): string {
+    return new BigNumber(this.estimateVsize())
+      .multipliedBy(this.targetFeeRate)
+      .toString();
+  }
+
+  get feeIncrease(): string {
+    return new BigNumber(this.requiredFee).minus(this.currentFee).toString();
+  }
+
+  // Methods
+  /**
+   * Sets a new target fee rate for the RBF transaction.
+   *
+   * @param {FeeRateSatsPerVByte} feeRate - The new target fee rate
+   * @returns {RbfTransaction} A new instance with the updated fee rate
+   */
+  public setTargetFeeRate(feeRate: FeeRateSatsPerVByte): RbfTransaction {
+    //The spread operator { ...this } doesn't work as intended in this context because
+    // the RbfTransaction instance doesn't have all the properties required by RbfTransactionOptions
+    const newInstance = new RbfTransaction({
+      psbt: this.originalPsbt,
+      network: this.network,
+      targetFeeRate: feeRate,
+      feeOutputIndex: this.feeOutputIndex,
+      dustThreshold: this.dustThreshold.toNumber(),
+      additionalUtxos: this.additionalUtxos,
+      requiredSigners: this.requiredSigners,
+      totalSigners: this.totalSigners,
+      changeOutputIndices: this.changeOutputIndices,
+    });
+    newInstance._targetFeeRate = feeRate;
+    return newInstance;
+  }
```

Comment:
this isn't right and is probably why you were having issues. a method on an instance of a class should be returning a new instance of the class unless you meant to have this be a [static method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) to instantiate a new instance? I don't think you did though. 

if all you're trying to do is set the fee rate, then just do this._targetFeeRate = feeRate.

Reply:
Thanks for the correction , there were a lot of these flaws in the current RBF class that I had to think over and hence so much delay .... But now the addition of the state management (using the caching mechanism ) has removed most of the flaws I feel ... 

changes done [here](https://github.com/caravan-bitcoin/caravan/pull/114/commits/4f4f97a08243f63ca31c4f0fc34216671e97e1de#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR104)